### PR TITLE
Unstoppable domain resolution

### PR DESCRIPTION
**Pull request by JohnnyJumper**
Implemented .zil naming resolution using the namicorn library that supports both .eth and .zil names:
* Replaced existing ens library calls with namicorn
* Renamed classes and methods to be abstract from the naming service used

### CODE REVIEW

File: ui/app/pages/send/send-content/add-recipient/naming-input.component.js [hunk 1]
```diff
@@ -45,14 +40,12 @@ export default class EnsInput extends Component {
 
   componentDidMount () {
     const network = this.props.network
-    const networkHasEnsSupport = getNetworkEnsSupport(network)
-    this.setState({ ensResolution: ZERO_ADDRESS })
+    this.namicorn = new Namicorn({blockchain: {ens: {network: parseInt(network)}, zns: true}})
```
Comment (balthazar) [inline ui/app/pages/send/send-content/add-recipient/naming-input.component.js]: Missing some spaces here for styling.

File: ui/app/helpers/utils/util.js [hunk 2]
```diff
@@ -2,6 +2,7 @@ const abi = require('human-standard-token-abi')
 const ethUtil = require('ethereumjs-util')
 const hexToBn = require('../../../../app/scripts/lib/hex-to-bn')
 import { DateTime } from 'luxon'
+// import Namicorn from 'namicorn'
```
Comment (balthazar) [inline ui/app/helpers/utils/util.js]: Could remove this comment

File: ui/app/pages/send/send.selectors.js [hunk 3]
```diff
@@ -248,7 +248,7 @@ function getTokenBalance (state) {
 }
 
 function getSendEnsResolution (state) {
```
Comment (balthazar) [inline ui/app/pages/send/send.selectors.js]: Was it a choice to keep the old ens naming?

File: ui/app/pages/send/send.container.js [hunk 4]
```diff
@@ -128,10 +127,11 @@ function mapDispatchToProps (dispatch) {
     qrCodeDetected: (data) => dispatch(qrCodeDetected(data)),
     updateSendTo: (to, nickname) => dispatch(updateSendTo(to, nickname)),
     fetchBasicGasEstimates: () => dispatch(fetchBasicGasEstimates()),
-    updateSendEnsResolution: (ensResolution) => dispatch(updateSendEnsResolution(ensResolution)),
+    updateSendEnsResolution: (namingResolution) => dispatch(updateSendEnsResolution(namingResolution)),
     updateSendEnsResolutionError: (message) => dispatch(updateSendEnsResolutionError(message)),
-    updateToNicknameIfNecessary: (to, toNickname, addressBook) => {
-      if (isValidENSAddress(toNickname)) {
+    updateToNicknameIfNecessary: (to, toNickname, addressBook, network) => {
+      const namicorn = new Namicorn({blockchain: {ens: {network}, zns: true}})
```
Comment (balthazar) [inline ui/app/pages/send/send.container.js]: Spaces here.

File: ui/app/pages/send/send.container.js [hunk 5]
```diff
@@ -67,6 +64,7 @@ module.exports = compose(
   connect(mapStateToProps, mapDispatchToProps)
 )(SendEther)
 
+
```
Comment (balthazar) [inline ui/app/pages/send/send.container.js]: Extra newline

File: ui/app/pages/send/send-content/add-recipient/naming-input.component.js [hunk 6]
```diff
@@ -66,37 +59,31 @@ export default class EnsInput extends Component {
     } = this.props
 
     if (prevProps.network !== network) {
-      const provider = global.ethereumProvider
-      this.ens = new ENS({ provider, network })
+      this.namicorn = new Namicorn({blockchain: {ens: {network: parseInt(network)}, zns: true}})
```
Comment (balthazar) [inline ui/app/pages/send/send-content/add-recipient/naming-input.component.js]: Spaces here

File: ui/app/pages/settings/contact-list-tab/add-contact/add-contact.component.js [hunk 7]
```diff
@@ -33,6 +35,8 @@ export default class AddContact extends PureComponent {
   constructor (props) {
     super(props)
     this.dValidate = debounce(this.validate, 1000)
+    this.namicorn = new Namicorn({blockchain: {ens: {network: parseInt(props.network)}, zns: true}})
```
Comment (balthazar) [inline ui/app/pages/settings/contact-list-tab/add-contact/add-contact.component.js]: Spaces

File: ui/app/pages/settings/contact-list-tab/add-contact/add-contact.component.js [hunk 8]
```diff
@@ -47,12 +51,15 @@ export default class AddContact extends PureComponent {
         }
       }
     }
+    if (this.props.network !== nextProps.network) {
+      this.namicorn = new Namicorn({blockchain: {ens: {network: parseInt(nextProps.network)}, zns: true}})
```
Comment (balthazar) [inline ui/app/pages/settings/contact-list-tab/add-contact/add-contact.component.js]: Ditto

File: ui/app/pages/settings/contact-list-tab/add-contact/add-contact.component.js [hunk 9]
```diff
@@ -61,16 +68,16 @@ export default class AddContact extends PureComponent {
...
         scanQrCode={_ => { this.props.scanQrCode() }}
         onChange={this.dValidate}
         onPaste={text => this.setState({ ethAddress: text })}
         onReset={() => this.setState({ ethAddress: '', ensAddress: '' })}
-        updateEnsResolution={address => {
+        updateNamingResolution={address => {
           this.setState({ ensAddress: address, error: '', ensError: '' })
         }}
-        updateEnsResolutionError={message => this.setState({ ensError: message })}
+        updateNamingResolutionError={message => this.setState({ ensError: message })}
```
Comment (balthazar) [inline ui/app/pages/settings/contact-list-tab/add-contact/add-contact.component.js]: Still using `ensError` variable name

File: ui/app/pages/send/send-content/add-recipient/add-recipient.container.js [hunk 10]
```diff
@@ -16,11 +16,11 @@ import AddRecipient from './add-recipient.component'
 export default connect(mapStateToProps, mapDispatchToProps)(AddRecipient)
 
 function mapStateToProps (state) {
-  const ensResolution = getSendEnsResolution(state)
+  const namingResolution = getSendEnsResolution(state)
```
Comment (balthazar) [inline ui/app/pages/send/send-content/add-recipient/add-recipient.container.js]: See comment in `send.selectors.js`

**Comment by whymarrh** [comment]
Hey, @JohnnyJumper and @bogdan (and the rest of the UD team) thanks for taking the time to work on this. I know that our teams have communicated about this integration—ultimately this isn't something we want to merge right now. We are happy with the feature-set of ENS right now and are looking to expand our support for it. There's definitely space to reconsider this integration in the future (possibly via a MetaMask Snap?) but I don't want to leave this open to collect merge conflicts in the meantime.

The Snaps route might make more sense as MetaMask could prompt the user to install that Snap if it detects an UD address. This is sometime to explore in the future.

**Pull request closed without merging**